include/swift/AST/DiagnosticsSema.def

```diff
@@ -664,8 +664,8 @@ ERROR(expr_keypath_mutating_getter,none,
       "which has a mutating getter",
       (const ValueDecl *, bool))
 ERROR(expr_keypath_static_member,none,
-      "%select{key path|dynamic key path member lookup}1 cannot refer to static member %0",
-      (const ValueDecl *, bool))
+      "static member %0 cannot be used on instance of type %1",
+      (const ValueDecl *, Type))
```

Comment:
We already have a diagnostic for this: `could_not_use_type_member_on_instance`

Reply:
But we also need to bring back old diagnostic and use it if the experimental feature flag is not set since in that mode static members cannot actually be referenced.

Reply:
I have updated the diagnostic so that the if the experimental flag is set, `could_not_use_type_member_on_instance` is emitted, else the old diagnostic is used.